src/components/plugins/mapdon-ksv.js

```diff
@@ -0,0 +1,99 @@
+import Button from 'react-bootstrap/lib/Button';
+import {MapdonHKRPlugin} from './mapdon-hkr';
+import CommentDisclaimer from "../CommentDisclaimer";
+import Input from 'react-bootstrap/lib/Input';
+import React from 'react';
+import {injectIntl} from 'react-intl';
+
+
+class MapdonKSVPlugin extends MapdonHKRPlugin {
+  constructor(props) {
+    super(props);
+    this.pluginInstanceId = "ksv" + (0 | (Math.random() * 10000000));
+  }
+
+  render() {
+    const buttonDisabled = (this.submitting);
+    const pluginPurpose = this.props.pluginPurpose;
+    const commentBox = (
+      <div>
+        <br/>
+        <Input
+          type="textarea"
+          onChange={this.handleTextChange.bind(this)}
+          placeholder="Kommentoi ehdotustasi tässä."
+        />
+        <p>
+          <Button bsStyle="primary" onClick={this.getDataAndSubmitComment.bind(this)} disabled={buttonDisabled}>
+            Lähetä ehdotus
+          </Button>
+        </p>
+        <CommentDisclaimer/>
+      </div>
+    );
+    return (
+      <div className="plugin-comment-form mapdon-ksv-plugin-comment-form">
+        <form>
+          <iframe
+            src="/assets/mapdon-ksv/plugin-inline.html"
+            className="plugin-frame mapdon-ksv-plugin-frame"
+            ref="frame"
+          ></iframe>
+          {pluginPurpose === 'postComments' ? commentBox : null}
+        </form>
+      </div>
+    );
+  }
+
+  componentDidMount() {
+    const iframe = this.refs.frame;
+    const {data, pluginPurpose} = this.props;
+    let {comments} = this.props;
+    const self = this;
+    if (!comments) {
+      comments = [];
+    }
+    if (!self._messageListener) {
+      self._messageListener = this.onReceiveMessage.bind(self);
+      window.addEventListener("message", self._messageListener, false);
+    }
+
+    iframe.addEventListener("load", () => {
+      self.sendMessageToPluginFrame({
+        message: "mapData",
+        data,
+        pluginPurpose,
+        comments,
+        instanceId: self.pluginInstanceId
+      });
+    }, false);
+  }
+
+  componentWillUpdate() {
+    const {data, pluginPurpose} = this.props;
+    let {comments} = this.props;
+    const self = this;
```

Comment:
Hm, that's a good question. A remnant of copypaste, don't know why the original plugin had it either. :)